Project-1/README.md

```diff
@@ -63,60 +68,256 @@ T	T	F	T	T
+(c) A cousin is a relationship between two individuals whose parents are siblings. Such that you have a cousin in the child of your aunt
+
+(d) ∃x∃y∀z(P(y,z)→z=x) - There is some x and some y such that for all z if y is the parent of z then z and x are identical 
+(d) An only child is an individual who was raised in a house primarily without siblings. Individuals who have step or half siblings but were raised pimarily alone are only children. 
+
+(e) ∃v∃w∃x∃y∀z(P(v,x)∧P(v,w)∧¬Fw∧P(v,y)∧¬Fy∧∀z((P(v,z)∧¬Fz)→((z=x)∨(z=w)∨(z=y)))) - There is some v and some w and some x and some y such that for all z if v is the parent of x and v is the parent of w and w is not female and v is the parent of y and y is not female and for all instances of z v is a parent of z and z is not female ten z is identical with w or z is identical with w or z is identical with y
+(e) An individual has exactly two brothers if and only if they have two siblings who identify as male.
 
 4. Let V be a vocabulary of the attribute (concept) language with complements (ALC) consisting of a role name "parent_of" and a concept name "Male". Interpret parent_of as "x is a parent of y" and M as "x is male". Where possible define the following formulas in this vocabulary; where not possible, explain why: 
   ```(a)  B that says that x is a brother of y
   (b)  A that says that x is an aunt of y
   (c)  C that says that x and y are cousins 
   (d)  O that says that x is an only child
   (e)  T that says that x has exactly two brothers 
-```
 
-5. Select two formulas defined in ALC from question 4 to form the basis of a T-Box. Supplement this T-box with whatever other axioms you like, as well as an A-box, so that you ultimately construct a knowledge base K = (T,A). Provide a _model_ of K. This may be graphical or symbolic or both. 
+  ⊔ ⊓ ⊧ ⊭ ⊦ ⊬ ⊏ ⊐ ⊑ ⊒ C ¬ ≡ ≠ ≥ ≤
+  parent_of
+  m
+
+  (a)  B that says that x is a brother of y
+  (a)  B ≡ M ⊓ ∃parentOf^-. (∃parentOf≥2)
+  (a) assuming gender binary determined by chromosomes 
+
+  (b)  A that says that x is an aunt of y
+  (b)  B ≡ M ⊓ ∃parentOf^-. (∃parentOf.(≥22parentOf.(parentOfM ⊔ ¬M) ⊓ ∃parentOf≥2)
+  (b) assuming gender binary determined by chromosomes 
+
+  (c)  C that says that x and y are cousins 
+  (c)  C ≡ ∃parentOf^-.(∃parentOf^-.(≥22parentOf.(parentOf M ⊔ ¬M))
+
+  (d)  O that says that x is an only child
+  (d) $O ≡ ∃parentOf^-.(∃parentOf.≤1)
+
+  (e)  T that says that x has exactly two brothers 
+  (e)  T ≡ M ⊓ ∃parentOf^-.(≤3∃parentOf. M) ⊔ (¬M ⊓ ∃parentOf^-.(≤2∃parentOf^-.M ⊓ ≥ 2∃parentOf^-.M))
+
+5. Select two formulas defined in ALC from question 4 to form the basis of a T-Box. Supplement this T-box with whatever other axioms you like, as well as an A-box, so that you ultimately construct a knowledge base K = (T,A). Provide a _model_ of K. This may be graphical or symbolic or both.
+
+ TBox = {Brother ⊆ Male,		(TBox.1)
+	Aunt ⊆ ~Male}		(TBox.2)
+
```

Comment:
Hi, I was working on this part with Ali today and we may have a problem with using the inverse of parentof_ since parentof_ is a primitive and is not defined as anything, so its reverse could technically take any value.
Keeping this in mind and using Karl's idea of reverse concepts, we built some intermediate concepts and came up with this (perhaps we can discuss it at tomorrow's meeting):

  (a)  B that says that x is a brother of y

  Per (Person) ≡ M ⊔ ¬M
  p2 (Parent of 2) ≡ ∃parentof. ≥ 2 ⊓ ¬(∃parentof. = 1) (this condition is meant to avoid that the two persons are the same, but I am not entirely sure it works or whether there are better ways to do so)
  
  B ≡ M ⊓ ∃p2¯.Per

  (b)  A that says that x is an aunt of y
  
  ∃p2¯.parentof.parentof.Per

  A ≡ ¬M ⊓ (∃p2¯. (parentof. ≥ 2).(parentof.Per)) ⊔ (¬parentof. Per ⊓ ∃p2¯.parentof.parentof.Per)
  
  U(uncle) ≡ M ⊓ (∃p2¯. (parentof. ≥ 2).(parentof.Per)) ⊔ ¬parentof.Per 

  (c)  C that says that x and y are cousins

  gp2(Grandparent with at least two children which each have children) ≡ ∃parentof ≥ 2.parentof.Per

  C ≡ gp2¯.per

  But this would only describe one of the two cousins, i.e. "being a cousin of someone" or "having a cousin".
  We cannot define "being cousins" in description logic.

  (d)  O that says that x is an only child  

  O ≡ ¬ ∃p2¯.Per

  (e)  T that says that x has exactly two brothers 

  This cannot be done without a role name for brother, we only have a concept name for it.

Reply:
lol. I just this now